pkg/operator/ceph/disruption/machinelabel/reconcile.go

```diff
@@ -0,0 +1,95 @@
+package machinelabel
+
+import (
+	"context"
+	"time"
+
+	"github.com/coreos/pkg/capnslog"
+	mapiv1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	"github.com/rook/rook/pkg/operator/ceph/disruption/controllerconfig"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	controllerName         = "machinelabel-controller"
+	MachineFencingLabelKey = "fencegroup.rook.io/cluster"
+)
+
+var log = capnslog.NewPackageLogger("github.com/rook/rook", controllerName)
+
+type ReconcileMachineLabel struct {
+	scheme  *runtime.Scheme
+	client  client.Client
+	log     *capnslog.PackageLogger
+	options *controllerconfig.Options
+}
+
+type machine struct {
+	isOccupiedByOSD bool
+	RawMachine      mapiv1.Machine
+}
+
+func (r *ReconcileMachineLabel) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.log.Infof("reconciling %v", request.NamespacedName)
+
+	// Fetch list of osd pods for the requested ceph cluster
+	pods := &corev1.PodList{}
+	err := r.client.List(context.TODO(), pods, client.InNamespace(request.Namespace), client.MatchingLabels{"app": "rook-ceph-osd", "rook_cluster": "rook-ceph"})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	machines := &mapiv1.MachineList{}
+	err = r.client.List(context.TODO(), machines, client.InNamespace("openshift-machine-api"))
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	nodeMachineMap := map[string]machine{}
+
+	r.log.Infof("Number of osd pods found %v", len(pods.Items))
+	r.log.Infof("Number of machines found %v", len(machines.Items))
+
+	// Adding machines to nodeMachineMap
+	for _, m := range machines.Items {
+		if m.Status.NodeRef != nil {
+			nodeMachineMap[m.Status.NodeRef.Name] = machine{RawMachine: m}
```

Comment:
What goes into the noderef? The `kubernetes.io/hostname` label or the `Node.ObjectMeta.Name`?

Reply:
`Node.ObjectMeta.Name`